File: app/src/main/java/com/suzei/racoon/ui/auth/login/LoginContract.java
package com.suzei.racoon.ui.auth.login;

import com.suzei.racoon.ui.base.Contract;

public interface LoginContract {

    interface View extends Contract.ProgressView {

        void setPresenter(Presenter presenter);

        void onLoginSuccess();

        void onLoginFailure(String message);

    }

    interface Presenter {

        void loginUser(String email, String password);

    }

}

File: app/src/main/java/com/suzei/racoon/ui/auth/login/LoginContract.kt
package com.suzei.racoon.ui.auth.login

import com.suzei.racoon.ui.base.Contract.ProgressView

interface LoginContract {
    interface View : ProgressView {
        fun setPresenter(presenter: Presenter?)
        fun onLoginSuccess()
        fun onLoginFailure(message: String?)
    }

    interface Presenter {
        fun loginUser(email: String?, password: String?)
    }
}